### PR TITLE
Format text custom function

### PR DESCRIPTION
**Pull request by subdavis**
For @zachmullen

### CODE REVIEW

**Comment by BryonLewis** [comment]
is this still too locked down?  Should you be able to edit the styling at the same time if you want to edit the text?  Text size, color and other options are still dependent on the type specified.  I don't know the immediacy of this or the intended usage, but just wondering if we should give some more control besides just formatting the TextData output.

**Comment by subdavis** [comment]
> text size, color and other options are still dependent on the type specified

You'd have to "coordinate" with the typeStyling properties, which I think is reasonable given that's how everything else works.

This is already much more than Zach needs, so I'm reluctant to go out of my way to make this ultra-flexible without a use case. This was just the most convenient way to expose the functions he needed.

**Review by BryonLewis — CHANGES_REQUESTED** [review]
Testing worked well.

I'm open to any ideas you may have about multiple text items per FrameDataTrack.  I just don't want to have to break or change this functionality afterwards to solve a VIAME issue.

File: client/src/layers/TextLayer.ts [hunk 1]
```diff
@@ -13,7 +13,43 @@ interface TextData {
...
+export type FormatTextRow = (track: FrameDataTrack) => TextData | null;
+
+interface TextLayerParams {
+  formatter?: FormatTextRow;
+}
+
+/**
+ * @returns value or null.  null indicates that the text should not be displayed.
+ */
+function defaultFormatter(track: FrameDataTrack): TextData | null {
```
[the remark this answers is not in the document]
Comment (subdavis) [reply client/src/layers/TextLayer.ts]: I looked at your changes from your other open PR, and returning an array here is perfectly fine with me.

64bc004

**Review by BryonLewis — APPROVED** [review]

**Pull request merged**